sky/serve/replica_managers.py

```diff
@@ -194,10 +198,19 @@ def _should_use_spot(task_yaml: str,
     return len(spot_use_resources) == len(task.resources)
 
 
-def with_lock(func):
+class HasLock(Protocol):
+    lock: threading.Lock
+
+
+L = typing.TypeVar('L', bound=HasLock)
```

Comment:
Can we just use `ReplicaManager` instead? It seems like `ReplicaManager` is the only thing that actually extends `HasLock`.